sklearn/metrics/pairwise.py

```diff
@@ -441,6 +441,45 @@ def pairwise_distances_argmin(X, Y, axis=1, metric="euclidean",
                                          batch_size=batch_size)[0]
 
 
+def haversine_distances(X, Y=None):
+    """Compute the Haversine distance between samples in X and Y
+
+    The Haversine distance is the angular distance between two points on
+    the surface of a sphere. The first distance of each point is assumed
+    to be the latitude, the second is the longitude, given in radians.
+    The dimension of the points must be 2.
```

Comment:
Maybe "dimension of the data"? A point doesn't have any dimensions by definition (it can be in an ND space though)...